pkg/log/zap/zap.go

```diff
@@ -154,10 +180,10 @@ func (o *Options) addDefaults() {
 	}
 
 	if o.Development {
-		if o.Encoder == nil {
-			encCfg := zap.NewDevelopmentEncoderConfig()
-			o.Encoder = zapcore.NewConsoleEncoder(encCfg)
+		if o.NewEncoder == nil {
+			o.Encoder = newConsoleEncoder(o.EncoderConfigOptions...)
```

Comment:
I think this needs to set the `NewEncoder`, not `Encoder`, because if `Encoder` is already set, we'll overwrite it here.

```suggestion
			o.NewEncoder = newConsoleEncoder
```